src/cmake_generator.cpp

```diff
@@ -611,13 +611,28 @@ void generate_cmake(const char *path, const parser::Project *parent_project) {
         const auto &packages = project.vcpkg.packages;
         for (size_t i = 0; i < packages.size(); i++) {
             const auto &package = packages[i];
-            if (!vcpkg_valid_identifier(package)) {
-                throw std::runtime_error("Invalid vcpkg package name '" + package + "'");
+            if (!vcpkg_valid_identifier(package.name)) {
+                throw std::runtime_error("Invalid vcpkg package name '" + package.name + "'");
             }
-            ofs << "    \"" << package << '\"';
-            if (i + 1 < packages.size()) {
-                ofs << ',';
+            if (package.features.empty()) {
+                ofs << "    \"" << package.name << '\"';
+            } else {
+                ofs << "    {\n";
+                ofs << "      \"name\": \"" << package.name << "\",\n";
+                ofs << "      \"features\": [";
+                for (size_t j = 0; j < package.features.size(); j++) {
+                    const auto &feature = package.features[j];
+                    ofs << '\"' << feature << '\"';
+                    if (j + 1 < package.features.size()) {
+                        ofs << ',';
+                    }
+                }
+                ofs << "]\n";
+                ofs << "    }";
             }
+			if (i + 1 < packages.size()) {
```

Comment:
I think here something went wrong with the formatting.

Reply:
I'm not quite sure I understand. This adds a trailing comma after the package if it's not the last package in the vector of packages. This is required in both cases (package with or without features). For example:
```
"dependencies": [
    {
      "name": "imgui",
      "features": ["docking-experimental","freetype","sdl2-binding","opengl3-binding"]
    },
    "fmt",
    "pegtl",
    "nativefiledialog",
    "spdlog",
    "utfcpp",
    "nlohmann-json",
    {
      "name": "glad",
      "features": ["gl-api-30"]
    }
  ],
```
Note the trailing comma after the object for `imgui`, as well as all the strings for the packages without features.

Reply:
Yeah, I should have specified. The C++ code seems to be indented incorrectly on GitHub (tabs vs spaces?)

Reply:
Fixed in 66621a9818fb6d75784f2959063869d2a0ea5091. Don't know how that happened. I have github set to show tabs as 4 spaces so it wasn't clear to me what the issue was. 🤦‍♀️

Reply:
Oh boy, I also fixed it and pushed ^^

Reply:
I think the problem was related to new VS2022 thinking I wanted to go back to tabs for some reason. Thanks for your patience.